### ISSUE TITLE
Files show as modified (orange) despite clean repository

### ISSUE THREAD

**Issue by willdady** [atom/tree-view#213]
I have an annoying problem where most files in my tree-view are shown as modified (orange). It's not clear to me why Atom thinks these files are modified. Doing a `git status` in the root shows there are no modified files and that everything is up to date with HEAD.

Also opening one of theses 'modified' files in Atom and then doing a Ctrl-Alt-Z has no effect, the file remains orange.

This seems like it **could** be related to issue #177 though I'm not using any submodules.

Running Atom v0.124.0 on Windows.

**Comment by trkoch** [comment]
I had a similiar issue. In my case the fix was to remove umlauts from file names.

**Comment by izuzak** [comment]
> I still think this is something Atom should report on better. Atom reporting the file as modified but nowhere explaining it's only a difference is filemodes is quite frustrating and makes it look like a bug with Atom.

@willdady Not sure if you even remember this issue :smile: but just wanted to ask in what way you'd expect/like that information to be presented (that the mode of a file was changed and that this is causing the file to be marked in the tree view as modified)? Thanks!

**Comment by willdady** [comment]
@izuzak Yes I do remember this issue. I really don't have a very strong opinion on where it should go. I guess it's more an exercise in UX. Here are a few suggestions though:

* Show an icon at the very top of the gutter? Maybe a white M on a red circle? (m = mode change)
* Atom shows an orange declaration warning icon in the status bar which has a tooltip on hover. Maybe keep with this pattern by having a different icon for 'version changes' this could give an overview of changes to the file, num lines added/removed, mode changes etc.

**Comment by izuzak** [comment]
Thanks, @willdady - yeah, I agree it's a UX thing, that's why asking users about expectations helps. We appreciate the thoughts and suggestions. :bow:

**Comment by timdp** [comment]
On Windows 8, I have `core.autocrlf` set to `true`, `core.filemode` to `false`, and Atom still shows most unchanged files as modified, right after a `git clone` and even after a `git reset --hard`.

Interestingly, it doesn't do it for binaries, so it's most likely still a CRLF issue. The actual files do have CRLFs in them as expected and `git status` doesn't list any changes.

Could it have something to do with the fact that I use [Babun](http://babun.github.io/), which comes with Git 2.1.4?

**Comment by timdp** [comment]
Oh, and no VirtualBox or anything like that. Just a regular NTFS volume with `C:\src\[repo]` as the root and 7-bit ASCII paths.

**Comment by timdp** [comment]
I also ran [this nodegit example](https://github.com/nodegit/nodegit/blob/master/examples/status.js) in my affected repository to check if this is a libgit2 issue. It didn't report any modified files.

**Comment by timdp** [comment]
I just found a fix for my situation: adding a `.gitattributes` with `* text=auto` and restarting Atom keeps it from showing the files as modified. Adding `eol=crlf`, on the other hand, didn't help.

I also tried creating `$HOME/.config/git/attributes` with the same contents, but that didn't change anything. Could be that I did something wrong though.

I guess I'll just add a `.gitattributes` to every repository I set up from now on. It's a good habit anyway. Note, however, that Atom's behavior still differs from that of stock Git, so it still looks to me like _something_ should be fixed.

Sorry for spamming this thread, but it appears that there are a couple of suggested fixes and none of them is a sure-fire one.

**Comment by davidspiess** [comment]
Thanks @timdp `* text=auto` worked for me too.

**Comment by raideer** [comment]
`* text=auto` works!

**Comment by kek91** [comment]
I got the same issue after changing permissions on the parent dir (which inherited permissions to the repo dir). For the sake of simplicity I just tried to clone the repo again and the problem solved itself.

**Comment by akekaluck** [comment]
On Window 7. I add `.git/info/attributes` with `* text=auto` worked for me.

**Comment by mrmckeb** [comment]
Happens on Windows 10 too, with Atom 1.1.0.
`git version 2.6.2.windows.1`

**Comment by MattHodge** [comment]
@mrmckeb +1 happening with same versions (win10, atom 1.1.0, git 2.6.2.windows.1)

**Comment by mrmckeb** [comment]
@MattHodge the `* text=auto` trick (above) worked for me, as a temporary workaround.

**Comment by 50Wliu** [comment]
For those whose issue got solved by using `* text=auto`, please see atom/atom#9424, which is an issue created specifically for that.

**Comment by MattHodge** [comment]
This is a global workaround I just found.. has to do with line endings in Windows with git https://help.github.com/articles/dealing-with-line-endings/#platform-windows

Just do:
`git config --global core.autocrlf true`

**Comment by mrmckeb** [comment]
Thanks @MattHodge, works great.

**Comment by pcantalupo** [comment]
I just installed atom 1.2.4 on windows 7 x64. I cloned a repo from github in cygwin terminal and did not make any changes to the repo. I am getting same issue where unaltered files are highlighted in orange. I noticed that the files highlighted orange are executable files. All other files are not highlighted orange.

I tried @MattHodge [fix](https://github.com/atom/tree-view/issues/213#issuecomment-154768405) of `git config --global core.autocrlf true` but it didnt' change anything. 

I also tried adding a `./.gitattributes` file with contents `* text=auto`. After restarting Atom, files are still highlighted orange.

What should I try next?

**Comment by lee-dohm** [comment]
> I noticed that the files highlighted orange are executable files.

@pcantalupo When you do `git status` what does it say?

**Comment by pcantalupo** [comment]
Branch was freshly cloned and status shows up-to-date with master:

```
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean

$ git --version
git version 2.6.2
```

If it helps, the repo that I'm using is [bioperl.github.io]( https://github.com/bioperl/bioperl.github.io) but is also occurring with other repos as well.

**Comment by lee-dohm** [comment]
And you're only seeing this behavior with executable files?

**Comment by pcantalupo** [comment]
yes

I performed a test on an executable file (`css/main.scss`) in bioperl.github.io by changing perms to `644` and committing. I restarted atom and the filename color changed to gray. Then, I changed perms to `755` and committed. After restarting atom, the filename color is orange. Would you like me to try anything else?

**Comment by lee-dohm** [comment]
`css/main.scss` isn't what most would call an executable file, even if it has the execute permission on it. And this is on a Windows machine using Cygwin? So it doesn't really have `644` or `755` permissions ... that's just Cygwin's translation of the actual Windows ACL, I assume.

**Comment by pcantalupo** [comment]
Yes, I'm on Win 7 x64 and running cygwin. I do not know how Cygwin handles permissions. All I know is that no matter what repo I'm editing in Atom, any executable file shows as orange despite it not being changed. Also, don't forget the test that I did in Cygwin. When I switched perms with `chmod 644 css/main.scss`, the file changed from orange to gray. But then it turned back to orange when I changed perms to 755 with `chmod 755 css/main.scss`. (If you don't like the `css/main.scss` example, I have a list of 65 executable Perl files from the `bioperl/bioperl-live` repo that do the same thing.)

I enjoy working in Atom on Mac and want to work with Atom on Windows but this issue is very annoying since many files look changed when they are not. I appreciate your help.

**Comment by lee-dohm** [comment]
What happens when you clone the repository on Windows and launch Atom *without* using Cygwin?

**Comment by pcantalupo** [comment]
I installed [Git](https://git-scm.com/download/win) for windows using installer defaults. I cloned bioperl.github.io with Git Windows and added it as project folder to Atom. There is no problem with the coloring. I noticed that the files all have '\r\n' line endings which I don't want on my system. So, how do I make Atom work with cygwin git cloned repos? If Atom does not support cygwin git cloned repos, my issue is resolved. Thank you

**Comment by 50Wliu** [comment]
If you change the files to just have `\n` line endings does the highlighting work properly with Cygwin (or vice-versa)?  You can do that through the Line Ending Selector on the right-hand side of the status bar.

**Comment by pcantalupo** [comment]
My cygwin git cloned repos already have `\n` line endings.

I tried toggling the line endings through the Selector for both the cygwin git and windows git repos but the coloring stayed the same.

**Comment by Elijen** [comment]
I have the same issue on OS X 10.10.4 and Atom v1.5.4. I can't really see any special pattern in what files are marked as modified. Is there a way how to see what git command and git executable is used by Atom to figure out the file state?

**Comment by CMCDragonkai** [comment]
I'm on Windows and using Cygwin git. Atom shows yellowish/orangish colour for certain folders even though nothing has changed. When I run `chmod 600` on the files inside the folder, then commit and push, Atom then shows standard grey colour.

**Comment by morcegon** [comment]
Same here. I'm using Atom 1.15.0 on MacOS Sierra. Nothing to commit but the files still orange (or yellow). 

![screen shot 2017-03-24 at 18 51 02](https://cloud.githubusercontent.com/assets/31614/24314923/e17782c2-10c2-11e7-9a28-50b066326111.png)

Nothing to commit in this file, but the icon of changes is there
![screen shot 2017-03-24 at 18 51 08](https://cloud.githubusercontent.com/assets/31614/24314924/e17b8d4a-10c2-11e7-850b-6f8a90128a8a.png)

**Comment by korpa** [comment]
Hi all,

I digged into this problem and realized, that git from Cygwin and the official Windows git from git-scm.com seem to be incompatible:

Cygwin:
```
$ ls -l
total 0
-rwxr-xr-x+ 1 user Domain Users 0 Sep 20 17:11 README.md

$ git version
git version 2.14.1

$ git status
On branch master
nothing to commit, working tree clean
```
git-scm git (mingw64) [git-bash]:
```
$ ls -l
total 0
-rw-r--r-- 1 user 1049089 0 Sep 20 17:11 README.md

$ git version
git version 2.14.1.windows.1


$ git status
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   README.md

no changes added to commit (use "git add" and/or "git commit -a")

```

Now I searched for git.exe in the atom directory and found a git.exe in ```resources/app/node_modules/dugite/git/cmd/git.exe``` and ```./resources/app/node_modules/dugite/git/mingw64/bin/git.exe```. I replaced these two git.exe with the git.exe from cygwin and voila, the files in "Unstaged changes" in the git sidebar have gone:

![image](https://user-images.githubusercontent.com/2982785/30652337-6d21f58a-9e28-11e7-9f5b-ec95dcb34876.png)

Sadly this didn't work for tree-view. 

It seems to me that there is another mingw64 git.exe somewhere, but I still was not able to find it. I checked ```process.env.PATH``` variable if there is another Git directory in the PATH but with no luck.

* Does anyone know how tree-view determines which git.exe to use? 
* Does tree-view even use a git.exe?

**Comment by korpa** [comment]
@morcegon Copying the cygwin git.exe to the "dugite" node module fixed the "change icon" for me as well:
![image](https://user-images.githubusercontent.com/2982785/30653263-ea1d0aa0-9e2a-11e7-9941-fc5202c1d20a.png)

**Comment by kbrannen** [comment]
@Korpa That doesn't work for me. I think I tried all the variations of both (like you did) and 1 or the other.

I will note that there is also a git-wrapper.exe that might help you ... maybe (it didn't help me):

```
$ ll cmd/git.exe mingw64/share/git/git-wrapper.exe
-rwx------+ 1 kbrannen Domain Users 38552 Nov 13 11:53 cmd/git.exe
-rwx------+ 1 kbrannen Domain Users 38552 Nov 13 11:46 mingw64/share/git/git-wrapper.exe
$ cksum cmd/git.exe mingw64/share/git/git-wrapper.exe
1519214228 38552 cmd/git.exe
188875033 38552 mingw64/share/git/git-wrapper.exe
```

So same size but apparently different contents. Swapping the cygwin git.exe there doesn't help me either.

I also note that atom's git is "2.15.0.windows.1" while cygwin's is 2.14.2; but I have trouble imagining that's enough of a diff to matter. I could see file path issue perhaps, but I thought the cygwin layer was good enough to deal with that (perhaps not).

I guess I'll just not use the git panel for now, which removes some of the usefulness of Atom. :( I'll try again later in a month or two.

Kevin

**Comment by tvld** [comment]
I got this issue today after renaming a module... weird. not on Windows but Ubuntu though.
Tree view shows all files in submodule folder are changed, but git status is nothing to commit

**Comment by kev153** [comment]
I have the same issue on Ubuntu (all files yellow but nothing to commit). The .gitattributes file didn't help. Could it be related to the fact that I moved the directory to a different location and/or changed attributes of the files (public read access)?

**Comment by tvld** [comment]
@kev153  might be... I don't have it since some time, but I remember moving folders as well....

**Comment by lee-dohm** [comment]
@kev153 Yes, changing the attributes of the files counts as a modification.

**Comment by kev153** [comment]
@lee-dohm Thanks, I didn't know that. Putting something like 
```
[core]
    filemode = false
```
into the `.git/config` file solves the issue.

**Comment by galrom** [comment]
In my case there were packages that were included in the git.ignore file - renaming these packages solved it.

**Comment by MoritzFS** [comment]
I just had the same issue happen, i.e. a completely clean repo but atom thinking it is not clean because of é in filename. Changing it from é->e fixed the issue. Atom 1.60.0 on mac.
